### ISSUE TITLE
Add ability to handle snapshots to vsphere_guest module

### ISSUE THREAD

**Issue by ansibot** [ansible/ansible#29640]
From @ronator on 2016-09-20T13:25:07Z

Original issue was https://github.com/ansible/ansible-modules-core/issues/2671. I used a wrong name in that ticket, I meant vsphere_guest. I was suggested by ansibot to re-open it here.

ansibot commented 7 minutes ago
@ronator: Thanks for taking the time to open this issue, but it was filed under the wrong github repository. Please create a new issue at https://github.com/ansible/ansible-modules-core/issues
##### ISSUE TYPE

- Feature Idea
##### COMPONENT NAME

vsphere_guest
##### ANSIBLE VERSION

```
ansible 2.1.1.0
```
##### OS / ENVIRONMENT

"N/A"
##### SUMMARY

The vmware_guest module lacks the ability to handle (list/create/remove) **snapshots** of vmware guests. This would be a killer feature! It would also allow more automization of the 'snapshot_to_clone' option, because for now using this option, snapshot must already be taken in vCenter.

Copied from original issue: ansible/ansible-modules-core#4942

**Comment by ansibot** [comment]
From @ansibot on 2016-09-20T13:25:07Z



@chrrrles, @rhoop, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
From @Abdul-Anshad-A on 2016-09-20T13:25:07Z

@ronator I have raised a pull request to handle snapshot operations with vsphere_guest and its pending on maintainer's review. Meanwhile, can you help me out with testing the changes ? You will find more instruction on pull request #5390 

https://github.com/Abdul-Anshad-A/ansible-modules-core/blob/snapshot_ops/cloud/vmware/vsphere_guest.py

**Comment by ansibot** [comment]
From @ronator on 2016-09-20T13:25:07Z

Sorry, I missed your message. I will read it and see if can help you. Embarrassingly, I do not know how to test it. I am relatively new to github. I cloned your branch "snapshots_ops" and know I am stuck :(
Anyway I have to say :+1: for the effort.

**Comment by ansibot** [comment]
From @Abdul-Anshad-A on 2016-09-20T13:25:07Z

@ronator No issues, Let me try to explain the testing process as simple as possible.

### 1)Install from source & setup environment

```
$ git clone git://github.com/ansible/ansible.git --recursive
$ cd ./ansible
$ source ./hacking/env-setup
```

### 2) Replace vsphere_guest module with the new modified one

`cp -f <path_to_vsphere_guest.py_cloned_from_my_snapshot_ops_branch> ./lib/ansible/modules/core/cloud/vmware/vsphere_guest.py`

### 3)Create play books accordingly as instructed on pull request #5390

Example:

```
vsphere_guest:
    vcenter_hostname: <hostname>
    username: <username>
    password: <password>
    validate_certs: False
    guest: <vm_name>
    snapshot_op:
        op_type: list_all
```

NOTE - Please, test it on your non-production environment.

**Comment by biomassives** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  acmeideal@gmail.com
    Domain biomassiv.es has exceeded the max emails per hour (101/100 (101%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
------ The body of the message is 5445 characters long; only the first
------ 5000 or so are included here.
Received: from github-smtp2-ext4.iad.github.net ([192.30.252.195]:60919 helo=github-smtp2a-ext-cp1-prd.iad.github.net)
	by chi-server32.websitehostserver.net with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.89)
	(envelope-from <noreply@github.com>)
	id 1drSqj-002QKg-UE
	for greg@biomassiv.es; Mon, 11 Sep 2017 12:50:22 -0500
Date: Mon, 11 Sep 2017 10:49:41 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1505152181;
	bh=8xN4uPGc24IC7DGUlP5aRc48H44DaypIhQI0deGXJM4=;
	h=From:Reply-To:To:Cc:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=rML6cK5FvtAMUnzGb3+RE+7JuochcFrQ65IffYjm+aUzr1qt6zA51KNkmWqeDET63
	 9EL4FD8WBUNZNydbukfxanDLFcrdVjg61jTHqUaVnYcsg702BBR8UWKFp4c5sgBK52
	 BRE2CfX1kPZzaFaiDXh2foJp+M6TyxC9dKalpp+Y=

**Comment by ronator** [comment]
I will try to come back to this upcoming week. Thanks for all your effort.

**Comment by cigamit** [comment]
https://github.com/ansible/community/wiki/VMware:-progress-tracker#deprecating-vsphere_guest
I believe once we gain full parity between vsphere_guest and vmware_guest is achieved, then this module will be deprecated.  We already have a separate module for create/list/deleting snapshots also.

**Comment by ansibot** [comment]
cc @akasurde @dagwieers @dav1x @jctanner @nerzhul
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by Akasurde** [comment]
Thanks for reporting this issue. I am closing this issue as there is a module to manage snapshots [vmware_guest_snapshot](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/vmware/vmware_guest_snapshot.py). Please feel free to re-open the issue.

**Comment by bhudlemeyer** [comment]
I know this is closed but I would like to say I stumbled across this when searching for a way to list all snapshots. The module referenced as the reason for this closer cant currently list all snapshots it can only modify or create them it seems.
